### PR TITLE
Refactor: Use github.com/spf13/cobra to execute cmd for apiserver

### PR DESCRIPTION
**Pull request by nuclearwu**
Signed-off-by: wuzhongjian <wuzhongjian_yewu@cmss.chinamobile.com>

### Description of your changes
refer to [kubesphere api server](https://github.com/kubesphere/kubesphere/blob/master/cmd/ks-apiserver/apiserver.go#L25-L32) and [kubevea plugin](https://github.com/kubevela/kubevela/blob/master/cmd/plugin/main.go#27-L35)
Use github.com/spf13/cobra to execute cmd.

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: cmd/apiserver/app/server.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+// NewAPIServerCommand creates a *cobra.Command object with default parameters
+func NewAPIServerCommand() *cobra.Command {
+	s := options.NewServerRunOptions()
+	cmd := &cobra.Command{
+		Use: "apiserver",
+		Long: `The KubeVela API server validates and configures data for the API objects. 
+The API Server services REST operations and provides the frontend to the
+cluster's shared state through which all other components interact.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 1 && args[0] == "build-swagger" {
```
Comment (barnettZQG) [inline cmd/apiserver/app/server.go:50]: Can we make the "build-swagger" a standalone command?

File: cmd/apiserver/app/server.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		go utils.EnablePprof(s.GenericServerRunOptions.PprofAddr, errChan)
+	}
+
+	go func() {
+		if err := run(ctx, s, errChan); err != nil {
+			errChan <- fmt.Errorf("failed to run apiserver: %w", err)
+		}
+	}()
+	var term = make(chan os.Signal, 1)
+	signal.Notify(term, os.Interrupt, syscall.SIGTERM)
```
Comment (barnettZQG) [inline cmd/apiserver/app/server.go:116]: Is it repeated with the `signals.SetupSignalHandler()`?

File: cmd/apiserver/app/server.go (hunk 1 above)
Comment (nuclearwu) [reply cmd/apiserver/app/server.go]: Of course, it has been done

File: cmd/apiserver/app/server.go (hunk 2 above)
Comment (nuclearwu) [reply cmd/apiserver/app/server.go:123]: yes, it has been done

**Review by Somefive — COMMENTED** [review]
Hmm, the apiserver in kubevela/kubevela is not a typical kube-apiserver based extension. I'm okay with the change but is there any special reason for doing so except for the code style? For example, any functional benefits by reusing the component-base lib?

**Review by Somefive — COMMENTED** [review]
I see. LGTM. /cc @barnettZQG What do you think?

**Comment by Somefive** [comment]
> /assign @wonderflow /assign @barnettZQG /assign @Somefive

@nuclearwu P.S. Suggest not using "/assign" directly. Let's use assign if this is a decided task we needs to move on or keep an eye on that. If you want to call the approvers to review, using "@" itself is enough. :)

**Comment by nuclearwu** [comment]
> 

ok，thanks

**Review by Somefive — APPROVED** [review]

**Comment by nuclearwu** [comment]
> > /assign @wonderflow /assign @barnettZQG /assign @Somefive
> 
> @nuclearwu P.S. Suggest not using "/assign" directly. Let's use assign if this is a decided task we needs to move on or keep an eye on that. If you want to call the approvers to review, using "@" itself is enough. :)

ok, I got it

**Comment by nuclearwu** [comment]
@barnettZQG

**Pull request merged**